buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -770,13 +770,25 @@ class BuildPlugin implements Plugin<Project> {
     }
 
     static void applyCommonTestConfig(Project project) {
-        project.tasks.withType(RandomizedTestingTask) {
+        project.tasks.withType(RandomizedTestingTask) {task ->
             jvm "${project.runtimeJavaHome}/bin/java"
             parallelism System.getProperty('tests.jvms', project.rootProject.ext.defaultParallel)
             ifNoTests System.getProperty('tests.ifNoTests', 'fail')
             onNonEmptyWorkDirectory 'wipe'
             leaveTemporary true
 
+            if (name != "test") {
```

Comment:
This might be worth a comment out here. It looks like it makes sure that `check` includes all testing tasks and that non-`test` tasks are configure the same way as `test`.

I wonder if we we should do the "`check` depends on testing task" outside of this if statement. That way we can consolidate all of that logic to one place.

I wonder if we want this classpath setup stuff at all. Are there test tasks that don't work this way? I don't *think* so, but I don't want to override them.

Reply:
We don't set it to anything other than what's there for test.
We do set it explicitly for rest integ test runner because there might not be a test task there, 
but we set it to the same thing. I agree there's more we can clean up here, but  this PR is probably not the right place for it.

Reply:
++